src/D2L.CodeStyle.Analyzers/Immutability/ImmutabilityContext.cs

```diff
@@ -40,12 +41,34 @@ internal sealed partial class ImmutabilityContext {
 			SpecialType.System_ValueType
 		);
 
-		private ImmutabilityContext( IEnumerable<ImmutableTypeInfo> extraImmutableTypes ) {
-			m_extraImmutableTypes = extraImmutableTypes
-				.ToImmutableDictionary(
-					info => info.Type,
-					info => info
-				);
+		private ImmutabilityContext(
+			ImmutableDictionary<INamedTypeSymbol, ImmutableTypeInfo> extraImmutableTypes,
+			ImmutableHashSet<ITypeParameterSymbol> conditionalTypeParamemters
+		) {
+			m_extraImmutableTypes = extraImmutableTypes;
+			m_conditionalTypeParameters = conditionalTypeParamemters;
+		}
+
+		/// <summary>
+		/// This function is intended to provide a new immutability context that considers the conditional type parameters
+		/// of the ConditionallyImmutable type being checked as immutable. It is important this is only used in those cases,
+		/// and not for instance while validating type arguments to an [Immutable] type parameter.

```

Comment:
Not used in the linked function, which keeps it conditional type parameters from being passed to strictly immutable ones as arguments.

https://github.com/Brightspace/D2L.CodeStyle/blob/8baddd99a261c8b0125ce743f69a4a4dfd81c2df/src/D2L.CodeStyle.Analyzers/Immutability/ImmutabilityAnalyzer.cs#L128

Reply:
It would be "nice" if this wasn't sound _only_ because of two different parts of analysis covering for each other, but this is pretty straight forward and doesn't require a concept like "Conditional" vs "Definite" immutability